src/test/java/core/basesyntax/service/impl/CsvFileReaderServiceImplTest.java

```diff
@@ -0,0 +1,88 @@
+package core.basesyntax.service.impl;
+
+import static org.junit.Assert.assertEquals;
+
+import core.basesyntax.service.FileReaderService;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+public class CsvFileReaderServiceImplTest {
+    private static final FileReaderService fileReaderService = new CsvFileReaderServiceImpl();
+    private static final String EMPTY_STRING = "";
+    private static List<String> expectedResult;
+    @Rule
+    public final ExpectedException exceptionRule = ExpectedException.none();
+
+    @Before
+    public void setUp() {
+        expectedResult = new ArrayList<>(List.of(
+                "type,fruit,quantity",
+                "b,banana,20",
+                "b,apple,100",
+                "s,banana,100",
+                "p,banana,13",
+                "r,apple,10",
+                "p,apple,20",
+                "p,banana,5",
+                "s,banana,50",
+                "s,orange,100",
+                "r,melon,20"));
+    }
+
+    @After
+    public void tearDown() {
+        expectedResult.clear();
+    }
+
+    @Test
+    public void readFromFile_nullFilePath_NotOk() {
+        exceptionRule.expect(NullPointerException.class);
```

Comment:
It's bad practice to expect NPE, lets add some `null` check in readFromFile() method. (same for other tests)

Reply:
I implemented this check with 'Objects.requireNonNull(filePath, "File path cannot be null")' instead of check If and RuntimeException. I thought NPE is more precise and specifies that method throws NPE in case of null rather than then some RuntimeException.
At least in this article recommends to use Objects.requireNonNull to intentionally throwing an exception on unwanted null parameters. 
What is your opinion about this?

Reply:
Якщо Objects.requireNonNull() викидає таку саму NPE яка б виникла у відсутність перевірки Objects.requireNonNull(), то яка в ньому суть?

Reply:
Якщо Objects.requireNonNull() викидає таку саму NPE яка б виникла у відсутність перевірки Objects.requireNonNull(), то яка в ньому суть? Якщо ми пропишемо свою кастомну перевірку на null, і додамо інформативне повідомлення про те, що саме null, нам буде лугше розібратися що пішло не так.